kernel/src/process_binary.rs

```diff
@@ -130,6 +130,9 @@ pub struct ProcessBinary {
     /// set if the process is checked by a credential checker and a specific
     /// credential was used to approve this process. Otherwise this is `None`.
     pub credential: OptionalCell<AcceptedCredential>,
+
+    /// Credential to vaildate if the app needs to be loaded
+    pub valid_app: bool,
```

Comment:
Instead of duplicating this information, we can just call `self.header.enabled()` if we need to know this